src/domain/vocabulary/command.ts

```diff
@@ -0,0 +1,28 @@
+import type { ReaderTaskEither } from 'fp-ts/ReaderTaskEither'
+import type { VocabularyPort } from './port'
+import type {
+  HTTPNetworkError,
+  NetworkRequestAbortedError,
+  NetworkUnspecifiedError
+} from '@/shared/error/network'
+import type { SerializationError } from '@/shared/error/serialize'
+
+export type Deps = {
```

Comment:
The type should be opaque, otherwise having a creation function doesn't provide any benefits, as it can be circumvented.

Reply:
You're absolutely right, I have declared Deps in the `dependencies` file as an opaque type with its constructor.